src/phases/query/MultipleGroupStagesOnComputedFields.yml

```diff
@@ -0,0 +1,84 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This file defines templates to use in workloads exercising multiple group stages on computed
+  fields.
+
+InsertDataTemplate:
+  Name: InsertData
+  Type: Loader
+  Threads: 4
+  Phases:
+    OnlyActiveInPhases:
+      Active: {^Parameter: {Name: "active", Default: [0]}}
+      NopInPhasesUpTo: {^Parameter: {Name: "nopInPhasesUpTo", Default: 5}}
+      PhaseConfig:
+        Repeat: 1
+        Database: {^Parameter: {Name: "database", Default: test}}
+        MultipleThreadsPerCollection: true
+        CollectionCount: 1
+        Collection: {^Parameter: {Name: "collection", Default: Collection0}}
+        DocumentCount: {^Parameter: {Name: "documentCount", Default: 1e6}}
+        BatchSize: {^Parameter: {Name: "batchSize", Default: 3000}}
+        Document:
+          time: {^IncDate: {start: {^Now: {}}, step: 1000}}
+          symbol: {^Cycle: {ofLength: {^Parameter: {Name: "numGroups", Default: 100}}, fromGenerator: {^RandomString: {length: 6}}}}
+          price: {^RandomDouble: {min: 100.0, max: 1000.0}}
+
+QueryTemplate:
+  Name: {^Parameter: {Name: "name", Default: MultipleGroupStagesOnComputedFieldsTemplate}}
+  Type: CrudActor
+  Database: &Database {^Parameter: {Name: "database", Default: test}}
+  Threads: {^Parameter: {Name: "threads", Default: 10}}
+  Phases:
+    OnlyActiveInPhases:
+      Active: {^Parameter: {Name: "active", Default: [0]}}
+      NopInPhasesUpTo: {^Parameter: {Name: "nopInPhasesUpTo", Default: 5}}
+      PhaseConfig:
+        Repeat: {^Parameter: {Name: "repeat", Default: 10}}
+        Database: *Database
+        Collection: {^Parameter: {Name: "collection", Default: Collection0}}
+        Operations:
+        - OperationMetricsName: {^Parameter: {Name: "operationsMetricsName", Default: MultipleGroupStagesOnComputedFieldsTemplate}}
+          OperationName: aggregate
+          OperationCommand:
+            Pipeline: [{$addFields:
+                          {"hourDiff": {$dateDiff: {"startDate": "$time", "endDate": "$$NOW", "unit": "hour"}}}},
```

Comment:
There's a risk that using `$$NOW` in the query will make its selectivity variable, depending on the timing of other phases of the test, which could make the measurements less stable.

I suggest loading `$time` with a fixed range of values and using a constant for `endDate` that will result in about 25% of documents getting filtered out.

My reasoning for 25% is that we should hopefully see some performance improvement by pushing down the `$match` and filtering out 25% of documents before they have to get converted to BSON. We don't want to lean on that _too_ heavily to get our performance improvement, though, because it won't be directly applicable to time series (which can use index bounds to skip most events that would be filtered out).

Reply:
Ah yes, that makes sense. Done!